Make AddRelatedManagers look for "objects" on parent model

RJPercival opened the pull request:
Previously, AddRelatedManagers would fail if a related model had inherited its `objects` field from a parent class. This would result in missing relation attributes. This is fixed by using `get()` instead of `names`; the former searches the MRO for the symbol, whereas the latter only looks for symbols declared directly on the class.

## Related issues

Closes #652

RJPercival committed:
Make AddRelatedManagers look for "objects" on parent model

Previously, AddRelatedManagers would fail if a related model had inherited
its `objects` field from a parent class. This would result in missing
relation attributes. This is fixed by using `get()` instead of `names`;
the former searches the MRO for the symbol, whereas the latter only looks
for symbols declared directly on the class.

sobolevn commented on tests/typecheck/fields/test_related.yml:
```diff
@@ -626,6 +626,153 @@
+                class TransactionLog(AbstractLog):
+                    transaction = models.ForeignKey(Transaction, on_delete=models.CASCADE, related_name="logs")
+
+                Transaction().logs
```
Can we please use `reveal_type()` here?

sobolevn replied:
The same for all other cases of `.logs` access 🙂

RJPercival replied:
I'm going to rewrite the tests to be consistent with the style of the other tests.

RJPercival replied:
I've trimmed it down to just a single test case that fails prior to this fix and passes afterwards. It's almost identical to `test_foreign_key_from_superclass_inherits_correctly` except that it defines `objects` on `LibraryEntity`.

sobolevn reviewed (commented):
Thanks a lot, let's see what CI will say.

RJPercival committed:
Add failing test for relation to model inheriting `objects`

Fails with:
```
pytest_mypy_plugins.utils.TypecheckAssertionError: Invalid output:
Expected:
  main:2: note: Revealed type is "myapp.models.MyUser*"
  main:3: note: Revealed type is "myapp.models.MyUser*"
  <45 (diff)
  <45 (diff)
Actual:
  main:2: note: Revealed type is "myapp.models.MyUser*"
  main:3: note: Revealed type is "myapp.models.MyUser*"
  main:6: error: "MyUser" has no attribute "book_set" (diff)
  main:6: note: Revealed type is "Any"          (diff)
  main:7: error: "MyUser" has no attribute "article_set" (diff)
  main:7: note: Revealed type is "Any"          (diff)
```

sobolevn reviewed (approved):
Thanks a lot for working on it! 👍
